plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -51,42 +53,47 @@ func parseDeploymentFromReplicaSet(name string) string {
 		return ""
 	}
 
-	if !stringInRuneset(suffix, kubeAllowedStringAlphaNums) {
+	if containUnexpectedRuneSet(suffix, deploymentUnexpectedRegEx) {
 		// Invalid suffix
 		return ""
 	}
 
 	return name[:lastDash]
 }
 
-// get the cronJob name by stripping the last dash following some rules
-// return empty if it is not following the rule
+// Get the cronJob name by stripping the last dash following by the naming convention: JobName-UnixTime
+// based on https://github.com/kubernetes/kubernetes/blob/c4d752765b3bbac2237bf87cf0b1c2e307844666/pkg/controller/cronjob/cronjob_controllerv2.go#L594-L596.
+// Before v1.21 CronJob in Kubernetes has used Unix Time in second; after v1.21 is a Unix Time in Minutes.
+
 func parseCronJobFromJob(name string) string {
 	lastDash := strings.LastIndexAny(name, "-")
+
+	//Return empty since the naming convention is: JobName-UnixTime, if it does not have the "-", meanings the job name is empty
 	if lastDash == -1 {
-		// No dash
 		return ""
 	}
+
 	suffix := name[lastDash+1:]
-	if len(suffix) != 10 {
-		// Invalid suffix if it is not 10 rune
+	suffixInt, _ := strconv.ParseInt(suffix, 10, 64)
+	//Convert Unix Time In Minutes to Unix Time
+	suffixStringMultiply := strconv.FormatInt(suffixInt*60, 10)
+
+	//Checking if the suffix is a unix time by checking: the length and contains character
+	// - Checking for the length: CronJobControllerV2 is Unix Time in Minutes (7-9 characters) while CronJob is Unix Time (10 characters).
+	//   However, multiply by 60 to convert the Unix Time In Minutes back to Unix Time in order to have the same condition as Unix Time
+	if len(suffix) != 10 && len(suffixStringMultiply) != 10 {
 		return ""
 	}
 
-	if !stringInRuneset(suffix, cronJobAllowedString) {
-		// Invalid suffix
+	// - Checking for unexpected character such as having characters others than numbers
+	if containUnexpectedRuneSet(suffix, cronJobUnexpectedRegex) || containUnexpectedRuneSet(suffixStringMultiply, cronJobUnexpectedRegex) {
 		return ""
 	}
 
 	return name[:lastDash]
 }
 
-func stringInRuneset(name, subset string) bool {
-	for _, r := range name {
-		if !strings.ContainsRune(subset, r) {
-			// Found an unexpected rune in suffix
-			return false
-		}
-	}
-	return true
+func containUnexpectedRuneSet(name string, unexpectedRegEx string) bool {
```

Comment:
Both variables here are a string, so the argument list should be collapsed like it was originally: `name, unexpectedRegEx string)`

Reply:
Reasonable. New things to learn in golang. Fixed it in newest commit